### PR TITLE
Fail when creating a Scope with a null-parent #1354.

### PR DESCRIPTION
**Pull request by Treehopper**
* Prevent subsequent lookup of values in the parent-scope, from failing.
* In case of failure: Preserve stack trace to uncover root cause.

See #1354 

- - -
name: Pull Request
about: Create a report to help us improve
title: ''
labels: Status:Discovery
assignees: ''

- - -

## Environment
**Liquibase Version**:

**Liquibase Integration & Version**: <Pick one: CLI, maven, gradle, spring boot, servlet, etc.>

**Liquibase Extension(s) & Version**: 

**Database Vendor & Version**:

**Operating System Type & Version**:

## Pull Request Type

* \[ ] Bug fix (non-breaking change which fixes an issue.)
* \[ ] Enhancement/New feature (non-breaking change which adds functionality)
* \[ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
A clear and concise description of the issue being addressed.  Additional guidance [here](https://liquibase.jira.com/wiki/spaces/LB/pages/1274904896/How+to+Contribute+Code+to+Liquibase+Core).

* Describe the actual problematic behavior.
* Ensure private information is redacted.

## Steps To Reproduce
List the steps to reproduce the behavior.

* Please be precise and ensure private information is redacted
* Include things like
  * Files used - sql scripts, changelog file(s), property file(s), config files, POM Files
  * Exact commands used - CLI, maven, gradle, spring boot, servlet, etc.

## Actual Behavior
A clear and concise description of what happens in the software **before** this pull request.

* Include console output if relevant
* Include log files if available.

## Expected/Desired Behavior
A clear and concise description of what happens in the software **after** this pull request.

## Fast Track PR Acceptance Checklist:

<

!--- If you're unsure about any of these, just ask us in a comment. We're here to help|width=200,height=183!

-->

* \[ ] Build is successful and all new and existing tests pass
* \[ ] Added [Unit Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1274937609/How+to+Write+Liquibase+Core+Unit+Tests)
* \[ ] Added [Integration Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1276608569/How+to+Write+Liquibase+Core+Integration+Tests)
* \[ ] Documentation Updated

## Need Help?
Come chat with us on our [discord channel](https://discord.com/channels/700506481111597066/700506481572839505)

- - -
#### Links
* Fixed Issue: [https://github.com/liquibase/liquibase/issues/1354](https://github.com/liquibase/liquibase/issues/1354) 
* Local Branch: [https://github.com/liquibase/liquibase/tree/Treehopper-issue-1354](https://github.com/liquibase/liquibase/tree/Treehopper-issue/1354)
* Unit Tests: [https://github.com/liquibase/liquibase/pull/1377/checks](https://github.com/liquibase/liquibase/pull/1377/checks)
* Integration Tests: (shown in Unit Tests link above)
* Functional Tests: [https://jenkins.datical.net/job/liquibase-pro/job/Treehopper-issue-1354/](https://jenkins.datical.net/job/liquibase-pro/job/Treehopper-issue-1354/)

#### Testing
* Steps to Reproduce: [https://jenkins.datical.net/job/liquibase-pro/job/Treehopper-issue%2F1354/](https://jenkins.datical.net/job/liquibase-pro/job/Treehopper-issue%252F1354/)
* Guidance: 
  * No behavior change, just throws useful exception for library code making invalid calls

#### Dev Verification
Code review and automated tests

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-636) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by Treehopper** [comment]
Not sure what codecov is trying to tell me. Please let me know if there is something I should fix.

**Comment by molivasdat** [comment]
Hi @Treehopper Thanks for creating this PR.
Here’s what happens next:

A member of the Liquibase team will take a look at your contribution and may suggest
changes, additional tests, or provide other feedback. The PR will be prioritized
according to our development and testing capacity.

We’ll let you know when it’s ready to move to the next step or if any changes are needed.

**Comment by nvoxland** [comment]
I pushed a merge conflict fix to your branch, along with moving your test to ScopeTest.

Marking this as NoManualTesting since it's a null-check that should only be hittable by libraries using Liquibase, not by our code.

**Review by nvoxland — APPROVED** [review]

File: liquibase-core/src/main/java/liquibase/Scope.java [hunk 1]
```diff
@@ -26,6 +26,8 @@
 import java.nio.charset.Charset;
 import java.util.*;
 
+import static java.util.Objects.requireNonNull;
```
Comment (Treehopper) [inline liquibase-core/src/main/java/liquibase/Scope.java:29]: Looks like this is now unused.

**Review by suryaaki2 — APPROVED** [review]

**Pull request merged**